src/Microsoft.DocAsCode.MarkdownLite/JsonTokenTreeRenderer.cs

```diff
@@ -3,20 +3,25 @@
 
 namespace Microsoft.DocAsCode.MarkdownLite
 {
-    using System.Collections.Immutable;
-
     public class JsonTokenTreeRenderer
     {
         #region Block
 
         public virtual StringBuffer Render(IMarkdownRenderer renderer, MarkdownNewLineBlockToken token, MarkdownBlockContext context)
         {
-            return this.Insert("NewLine", this.GetSize(), Type.Leaf);
+            return this.Insert($"NewLine_{token.SourceInfo.LineNumber}", this.GetSize(), Type.Leaf);
         }
 
         public virtual StringBuffer Render(IMarkdownRenderer renderer, MarkdownCodeBlockToken token, MarkdownBlockContext context)
         {
-            return this.Insert("Code(" + token.Lang + ")", this.GetSize(), Type.Leaf);
+            if (token.Lang != null)
+            {
+                return this.Insert($"Code({token.Lang})_{token.SourceInfo.LineNumber}_{this.escape(token.Code)}", this.GetSize(), Type.Leaf);
```

Comment:
OK. It should works. Or consider using a separator that has already been escaped.

---
In reply to: [75989068](https://github.com/dotnet/docfx/pull/631#discussion_r75989068) [](ancestors = 75989068,75988569)